### DISCUSSION TITLE
Reset Config files menu or button

### DISCUSSION THREAD

**Discussion by khturnings** [winder/Universal-G-Code-Sender#2665] [Ideas]
Looking through many issues and threads I see that often the recommendations are to delete config files and/or settings.

Can a button or menu item be added to automatically do this. If these files are locked because they are in use by the program a temporary file could be created and program auto restarts - looks for temporary file and deletes config files in advance of loading default values and recreating config files.

Add this info to wiki or pin to issues so people try this before raising new issues thus saving a lot of unnecessary posts.

**Comment by breiler** [comment]
This is not that simple to fix.

The settings (or more correctly the cache) that are getting corrupted are deeply rooted in the Netbeans API:s that we are using. In order to fix this we need to hook into the Netbeans startup process before it is processing the cache. I made an attempt at this here: https://github.com/winder/Universal-G-Code-Sender/blob/master/ugs-platform/ugs-platform-ugscore/src/main/java/com/willwinder/ugs/nbp/core/lifecycle/StartupOptionProcessor.java#L98

This seems to work fine for MacOSX and Linux, but not Windows.

NetBeans is giving us so much, but right now I am honestly quite fed up with trying to keep it flying. I need a couple of days to recharge and maybe look for a more long term solution after new years.

**Reply by khturnings** [reply]
Thanks for looking at this - I'm a old school programmer so not conversant with Java/Netbeans so was hoping it was an easy option - Was mainly thinking of you and the number of times you tell people to delete those files.
Enjoy your break and look forward to new things in the new year.
